servers/src/grin/dandelion_monitor.rs

```diff
@@ -41,120 +40,212 @@ pub fn monitor_transactions<T>(
 ) where
 	T: BlockChain + Send + Sync + 'static,
 {
-	debug!(LOGGER, "Started Dandelion transaction monitor");
+	debug!(LOGGER, "Started Dandelion transaction monitor.");
+
 	let _ = thread::Builder::new()
 		.name("dandelion".to_string())
 		.spawn(move || {
 			loop {
+				if stop.load(Ordering::Relaxed) {
+					break;
+				}
+
+				// this is the patience timer, we loop every n secs.
+				let patience_secs = 10;
```

Comment:
Maybe externalize this value ?

Reply:
Yeah - let me take a look at your PR and I can pull the config changes into this, unless you want me to merge this as-is and then you can open a PR to make the "patience timer" stuff more robust and configurable?

Reply:
Having it all in this PR would be nice. But I can do it if you want, no preference of my side. 
Also speaking of that, the current values are 10 seconds of patience and 5 minutes of embargo. Which led to approximately 1m30s for a stem transaction to get in the mempool. Are you conformable with that? Should we reevaluate them?